### PR TITLE
Image factory removal

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Not ready yet

File: dali/internal/event/common/thread-local-storage.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014 Samsung Electronics Co., Ltd.
+ * Copyright (c) 2016 Samsung Electronics Co., Ltd.
```
Comment (xkpe) [inline dali/internal/event/common/thread-local-storage.cpp:2]: 2017

File: dali/internal/event/common/thread-local-storage.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #define __DALI_INTERNAL_THREAD_LOCAL_STORAGE_H__
 
 /*
- * Copyright (c) 2015 Samsung Electronics Co., Ltd.
+ * Copyright (c) 2016 Samsung Electronics Co., Ltd.
```
Comment (xkpe) [inline dali/internal/event/common/thread-local-storage.h:5]: 2017

**Review by xkpe — COMMENTED** [review]
Since you're changing dates, it's 2017 now

File: dali/internal/event/images/resource-image-impl.cpp [hunk 3]
```diff
@@ -26,7 +26,7 @@
 #include <dali/public-api/object/type-registry.h>
 #include <dali/integration-api/debug.h>
 #include <dali/internal/event/common/thread-local-storage.h>
-#include <dali/internal/event/images/image-factory.h>
+#include <dali/integration-api/platform-abstraction.h>
```
Comment (xkpe) [inline dali/internal/event/images/resource-image-impl.cpp:29]: should be next to the other integration-api include

File: dali/internal/event/images/resource-image-impl.cpp [hunk 4]
```diff
@@ -76,18 +77,19 @@ ResourceImagePtr ResourceImage::New( const std::string& url, const ImageAttribut
...
-    image->mHeight = attributes.GetHeight();
-    image->mRequest = image->mImageFactory.RegisterRequest( url, &attributes );
-    image->mTicket = image->mImageFactory.Load( *image->mRequest.Get() );
-    image->mTicket->AddObserver( *image );
+    image->mParameters.size= ImageDimensions::FromFloatVec2( attributes.GetSize() );
+    image->mParameters.scalingMode = attributes.GetScalingMode();
+    image->mParameters.samplingMode = attributes.GetFilterMode();
+    image->mParameters.orientationCorrection = attributes.GetOrientationCorrection();
+    image->mUrl = url;
+    image->Reload(); // reuse code
```
Comment (xkpe) [inline dali/internal/event/images/resource-image-impl.cpp:92]: How many steps does a ResourceImage need to go through to be initialised and loaded? We should rethink this API.

**Pull request closed without merging**